Cmdline/Action/KSP.cs

```diff
@@ -496,47 +502,107 @@ private int SetDefaultInstall(DefaultOptions options)
+                dlcVersion = null;
+            }
+            else
+            {
+                dlcVersion = options.dlcVersion;
+            }
+
+            // Parse the choosen KSP version
             try
             {
-                log.Debug("Parsing KSP version");
                 version = KspVersion.Parse(options.version);
-                if (!version.IsBuildDefined)
+            }
+            catch (FormatException)
+            {
+                // Thrown if there is anything besides numbers and points in the version string or a different syntactic error.
+                User.RaiseError("Please check the version argument - Format it like Maj.Min.Patch[.Build] - e.g. 1.6.0 or 1.2.2.1622");
+                return Exit.BADOPT;
+            }
+
+            if (!version.IsMajorDefined || !version.IsMinorDefined)
+            {
+                User.RaiseError("Please enter the at least the version major and minor values in the form Maj.Min - e.g. 1.5");
+                return Exit.BADOPT;
+            }
+
+            if (!version.IsFullyDefined)
+            {
+                // Let the user select the sepcific version.
+                KspVersion[] knownVersions = new GameVersionProviders.KspBuildMap(new Win32Registry()).KnownVersions.ToArray();
+                System.Collections.Generic.List<KspVersion> possibleVersions = new System.Collections.Generic.List<KspVersion>();
+                foreach (KspVersion ver in knownVersions)
+                {
+                    if (!version.IsPatchDefined)
+                    {
+                        if (version.Major == ver.Major && version.Minor == ver.Minor)
+                        {
+                            possibleVersions.Add(ver);
+                        }
+                    }
+                    else
+                    {
+                        if (version.Major == ver.Major && version.Minor == ver.Minor && version.Patch == ver.Patch)
+                        {
+                            possibleVersions.Add(ver);
+                        }
+                    }
+                }
+                if (possibleVersions.Count == 0 )
                 {
-                    version = version.FindKnownVersion();
+                    User.RaiseError("Your specified version is not known to CKAN. Please enter a real KSP version.");
+                    return Exit.BADOPT;
                 }
-                if (version == null)
+                else if (possibleVersions.Count == 1)
                 {
-                    throw new ArgumentOutOfRangeException(nameof(version), "Your specified version is invalid.");
+                    version = possibleVersions.ElementAt(0);
+                }
+                else
+                {
+                    int choosen = User.RaiseSelectionDialog("The specified version is not unique, please select one:", possibleVersions.ToArray());
+                    if (choosen > 0 && choosen < possibleVersions.Count)
```

Comment:
Selecting 1 doesn't work; I think `RaiseSelectionDialog` returns input-1, so this should be `>= 0`:

```
ckan ksp fake fake12 fake12 1.5
1) 1.5.0.2332
2) 1.5.1.2335
Enter a number between 1 and 2 (To cancel press "c" or "n".):
1
Selection cancelled! Please call 'ckan ksp fake' again.
```

Reply:
Oh yes, did fix that for the rest of the  new code, but oversaw this one.